src/request/sign-transaction/SignTransaction.js

```diff
@@ -139,12 +164,16 @@ class BaseLayout {
     }
 
     run() {
-        // Go to start page
-        window.location.hash = BaseLayout.Pages.CONFIRM_TRANSACTION;
-        this._passphraseBox.focus();
-
         // Async pre-load the crypto worker to reduce wait time at first decrypt attempt
-        Nimiq.CryptoWorker.getInstanceAsync();
+        if (this._request.keyInfo.encrypted) {
+            Nimiq.CryptoWorker.getInstanceAsync();
```

Comment:
Have you tested this? Maybe transaction generation/signing/hashing also requires the CryptoWorker? I'm not sure.